### PR TITLE
Medical Rework P. 1 - Defib And Death Timer Rework

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Completely reworks how corpses slowly rot away and become permanently dead. Instead of corpses suddenly becoming unrevivable at 5 minutes, they will slowly accrue heart damage for 7 minutes until heartbreak. 

However, they are still technically revivable for three more minutes. This is because now you will also accrue 'integrity' damage over time, which cannot be fixed on the colony and requires one to go up to get surgery on the ship. 'Integrity' damage is taken over 9 minutes until it destroys the heart. When this happens, the brain will stop recieving oxygen and is on a one-minute-timer until permanent, irretrievable death.

Corpses can be 'frozen' with stasis bags or cryogenic pods to prevent further decay.

Colonial surgery tables have been replaced with a new, weaker version that can do almost everything but fix organ integrity and brain surgery. They can still fix normal organ damage.

If you have more organ integrity damage than heart damage, the organ integrity value will be used to calculate symptoms of heart damage. This can never cause you to die or fall unconscious by itself if you're alive.

Removed the unused variable 'tod'.

Removed all check_tod() procs and replaced them with is_revivable()

Refactored a lot of code around organs, though it's still repugnant.

Added a new property, narcoleptic. It reduces metabolization rate by 33% if resting/stunned, and by 66% if sleeping/unconscious, the value is improved with levels.

Peridaxon now metabolizes at 1 unit per 20 seconds, instead of the original 1 unit per 40. It now has the narcoleptic property at level 1, meaning to get the most out of peridaxon you must be incapacitated or intentionally resting.

JS is broken for now

# Explain why it's good for the game

> Completely reworks how corpses slowly rot away and become permanently dead. Instead of corpses suddenly becoming unrevivable at 5 minutes, they will slowly accrue heart damage for 7 minutes until heartbreak. 

This is something that is meant to do several things at once.

1. It allows marines to stay in the round longer.
Instead of becoming perma-dead at 5 minutes and being unable to ever play the game as a marine again unless you manage to get a faceless rando cryorine spawn, this increases the defib timer by roughly 40%. This will allow marines to be able to play the game more rather than just ending up as skulls on one mistake. In exchange...
2. It encourages and reinforces shipside medbay by making it vital to operations by curing integrity loss.
3. It encourages groundside medbay, even, by allowing ground surgeons to reduce, if not cure, the damage caused by heart decay.

It's a three-pronged feature that is intended to help out all sectors of play by allowing marines to play for longer at the cost of needing more downtime to mend their wounds. This should be bundled with improvements to dropship flight time.

> However, they are still technically revivable for three more minutes. This is because now you will also accrue 'integrity' damage over time, which cannot be fixed on the colony and requires one to go up to get surgery on the ship. 'Integrity' damage is taken over 9 minutes until it destroys the heart. When this happens, the brain will stop recieving oxygen and is on a one-minute-timer until permanent, irretrievable death.

This means that technically, if you rush medevac into surgery, you can save someone who is heartbroken from damage over time via replacing their heart (We can still do that right?), which could be a fun little coop minigame if the stars align for it.

Additionally, it means that you actually die from brain death, not your heart failing.

> Corpses can be 'frozen' with stasis bags or cryogenic pods to prevent further decay.

This will further increase effective revival time to help marines live longer, at the cost of everything else stated elsewhere.

**A lot of things need tuning here.** Dropship timing, the exact numbers and breakthrough values, etc. There are surely many values here that could be improved for a better experience.

> Colonial surgery tables have been replaced with a new, weaker version that can do almost everything but fix organ integrity and brain surgery. They can still fix normal organ damage.

Again, to encourage going up shipside rather than using the dirty, dusty colony's operating theatres that are filled with blood, grime, acid, and resin. They're still great for anything else though.

> If you have more organ integrity damage than heart damage, the organ integrity value will be used to calculate symptoms of heart damage. This can never cause you to die or fall unconscious by itself if you're alive.

This is so that fixing your heart doesn't mean that you're perfectly fine until organ integrity suddenly makes you perma-dead. There's still penalties!

> Added a new property, narcoleptic. It reduces metabolization rate by 33% if resting/stunned, and by 66% if sleeping/unconscious, the value is improved with levels.

> Peridaxon now metabolizes at 1 unit per 20 seconds, instead of the original 1 unit per 40. It now has the narcoleptic property at level 1, meaning to get the most out of peridaxon you must be incapacitated or intentionally resting.

With Peridaxon just being less useful, marines would be less eager to use it to stay in the fight, and rather use it to retreat to a place they can get healed up. By making it still slow-metabolizing if the marine is resting, the rate of metabolization will hopefully not impact patients but just those who are up and about. It is still usable for those who wish to do this, but they won't be able to fill a canteen and be done for the entire OP.

Also a new shitty toy for researchers, their favorite. Enjoy!

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
balance: Completely reworks how corpses slowly rot away and become permanently dead. Instead of corpses suddenly becoming unrevivable at 5 minutes, they will slowly accrue heart damage for 7 minutes until heartbreak. 
balance: However, they are still technically revivable for three more minutes. This is because now you will also accrue 'integrity' damage over time, which cannot be fixed on the colony and requires one to go up to get surgery on the ship. 'Integrity' damage is taken over 9 minutes until it destroys the heart. When this happens, the brain will stop recieving oxygen and is on a one-minute-timer until permanent, irretrievable death.
balance: Corpses can be 'frozen' with stasis bags or cryogenic pods to prevent further decay.
balance: Colonial surgery tables have been replaced with a new, weaker version that can do almost everything but fix organ integrity and brain surgery. They can still fix normal organ damage.
balance: If you have more organ integrity damage than heart damage, the organ integrity value will be used to calculate symptoms of heart damage. This can never cause you to die or fall unconscious by itself if you're alive.
code: Removed the unused variable 'tod'.
code: Removed all check_tod() procs and replaced them with is_revivable()
refactor: Refactored a lot of code around organs, though it's still repugnant.
balance: Added a new property, narcoleptic. It reduces metabolization rate by 33% if resting/stunned, and by 66% if sleeping/unconscious, the value is improved with levels.
balance: Peridaxon now metabolizes at 1 unit per 20 seconds, instead of the original 1 unit per 40. It now has the narcoleptic property at level 1, meaning to get the most out of peridaxon you must be incapacitated or intentionally resting.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
Peridaxon and stasis are untested

File: code/modules/organs/organ_internal.dm [hunk 1]
```diff
@@ -129,6 +187,50 @@
...
+	if(!M)
+		return
+	RegisterSignal(M, COMSIG_MOB_DEATH, PROC_REF(start_decaying))
+	RegisterSignal(M, COMSIG_HUMAN_REVIVED, PROC_REF(stop_decaying))
+
+/datum/internal_organ/heart/proc/start_decaying()
+	START_PROCESSING(SSprocessing, src) // our code is so fucking bad dude this is the best way to do this
+
+/datum/internal_organ/heart/proc/stop_decaying()
+	STOP_PROCESSING(SSprocessing, src) // our code is so fucking bad dude this is the best way to do this
```
Comment (harryob) [inline code/modules/organs/organ_internal.dm:204]: these both need SIGNAL_HANDLER i don't want dcs going to sleep

File: code/modules/organs/organ_internal.dm [hunk 2]
```diff
@@ -129,6 +187,50 @@
...
+
+/datum/internal_organ/heart/process()
```
Comment (harryob) [inline code/modules/organs/organ_internal.dm:206]: shouldn't this be using deltatime

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/death.dm [hunk 3]
```diff
@@ -58,6 +58,9 @@
 	if(!gibbed && species.death_sound)
 		playsound(loc, species.death_sound, 50, 1)
 
+	for(var/datum/internal_organ/organ in internal_organs)
+		SEND_SIGNAL(organ, COMSIG_MOB_DEATH)
```
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/modules/mob/living/carbon/human/death.dm:62]: wtf!

**Comment by cm13live** [comment]
This pull request has been mentioned on **CM-SS13**. There might be relevant details there:

https://forum.cm-ss13.com/t/2806-medical-reowrk-p-1-defib-and-death-timer-rework/814/1

**Comment by harryob** [comment]
fix ur pops bro

**Pull request closed without merging**